src/charm.py

```diff
@@ -139,7 +139,13 @@ def __init__(self, *args):
             self,
             relation_name="self-metrics-endpoint",
             jobs=self.self_scraping_job,
-            external_url=self._external_url,
+            refresh_event=[
+                self.on.update_status,
+                self.ingress.on.ready,
+                self.ingress.on.revoked,
+                self.on["ingress"].relation_changed,
+                self.on["ingress"].relation_departed,
+            ],
```

Comment:
Shall we document this?

```suggestion
            ],
            # We do not provide an `external_url` arg here because scarping is
            # by definition within model boundaries, i.e. reachable via fqdn.
```